src/main/java/uk/gov/laa/ccms/caab/service/LookupService.java

```diff
@@ -236,6 +244,112 @@ public Mono<CommonLookupDetail> getApplicationStatuses() {
+      final String matterType,
+      final Boolean isLead) {
+
+    final String categoryOfLaw = application.getCategoryOfLaw().getId();
+    final Boolean isAmendment = application.getAmendment();
+    final Boolean larScopeFlag = application.getLarScopeFlag();
+    final String applicationType = application.getApplicationType().getId();
+
+
+    return ebsApiClient.getProceedings(
+        categoryOfLaw,
+        matterType,
+        isAmendment,
+        larScopeFlag,
+        applicationType,
+        isLead);
+  }
+
+  /**
+   * Retrieves Client involvement types with detailed parameters.
+   *
+   * @param proceedingCode The proceeding code.
+   * @return A Mono containing the ClientInvolvementTypeLookupDetail or an error handler if an
+   *         error occurs.
+   */
+  public Mono<ClientInvolvementTypeLookupDetail> getClientInvolvementTypes(
+      final String proceedingCode) {
+    return ebsApiClient.getClientInvolvementTypes(proceedingCode);
+  }
+
+  /**
+   * Get a list of Client Involvement Types Common Values.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
+   */
+  public Mono<CommonLookupDetail> getClientInvolvementTypes() {
+    return ebsApiClient.getCommonValues(COMMON_VALUE_CLIENT_INVOLVEMENT_TYPES);
+  }
+
+  /**
+   * Retrieves the level of service types for a given proceeding.
+   * The level of service types are fetched based on the proceeding code, category of law, and
+   * matter type.
+   *
+   * @param application The application detail object which contains the category of law.
+   * @param proceedingCode The code of the proceeding.
+   * @param matterType The type of the matter.
+   * @return A Mono of LevelOfServiceLookupDetail containing the level of service types.
+   */
+  public Mono<LevelOfServiceLookupDetail> getProceedingLevelOfServiceTypes(
+      final ApplicationDetail application,
+      final String proceedingCode,
+      final String matterType) {
+
+    final String categoryOfLaw = application.getCategoryOfLaw().getId();
+
+    return ebsApiClient.getLevelOfServiceTypes(proceedingCode, categoryOfLaw, matterType);
+  }
+
+  /**
+   * Get a list of Order Type Common Values.
+   *
+   * @return CommonLookupDetail containing the common lookup values.
+   */
+  public Mono<CommonLookupDetail> getOrderTypes() {
+    return ebsApiClient.getCommonValues(COMMON_VALUE_PROCEEDING_ORDER_TYPE);
+  }
+
+  /**
+   * Get the order type description, filtered by code.
+   *
+   * @return String containing the description of an order type.
+   */
+  public Mono<String> getOrderTypeDescription(final String code) {
+    return ebsApiClient.getCommonValues(COMMON_VALUE_PROCEEDING_ORDER_TYPE)
+        .map(commonLookupDetail -> commonLookupDetail.getContent().stream()
+            .filter(commonLookupValueDetail -> commonLookupValueDetail.getCode().equals(code))
+            .findFirst()
+            .map(CommonLookupValueDetail::getDescription)
+            .orElse(null));
```

Comment:
Could default to the passed in code, rather than returning null

Reply:
amended